File: Data/Models/Post.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AngularCore.Data.Models
{
    public class Post : BaseEntity
    {
        public string AuthorId { get; set; }

        [Required]
        public User Author { get; set; }

        public string WallOwnerId { get; set; }

        [Required]
        public User WallOwner { get; set; }

        public List<Comment> Comments { get; set; } = new List<Comment>();

        [Required]
        public virtual string Content { get; set; }
    }
}

File: Monolith/ClientApp/src/app/models/post.ts
import { User } from '@app/models/user';
import { Comment } from '@app/models/comment';

export class Post {

  constructor(
    public id: string,
    public author: User,
    public wallOwner: User,
    public comments: Comment[],
    public content: string,
    public modifiedAt: Date,
    public createdAt: Date
  ){}

}
